docs/design/2021-06-23-restricted-read-only.md

```diff
@@ -0,0 +1,45 @@
+# Technical Design Document for Cloud Read Only
+
+> Author: [huzhifeng@pingcap.com](mailto:huzhifeng@pingcap.com)
+This document is open to the web, and it will appear as a design doc PR to the TiDB repository.
+
+# Introduction
+
+To allow Cross-Region Replication on TiDB Cloud, we need a more strict level of read-only. In this document, we introduced the `TIDB_RESTRICTED_READ_ONLY` global variable. Turning it on will make the cluster read-only eventually for all users, including users with SUPER or CONNECTION_ADMIN privilege.
+
+# Motivation or Background
+
+See the [original specification](https://docs.google.com/document/d/1LtmW4fIohTjPcupD9smQGQpN2B2tRSFNyH8PFbg-L2c/edit).
+
+# **Detailed Design**
+
+## read-only handling
+
+Currently, TiDB supports `read_only` and `super_read_only` global variables with no actual operation. Since TIDB_RESTRICTED_READ_ONLY does not comply with the behavior of MySQL read-only and super-read-only (which will block if there is table locked or ongoing committing of a transaction), we will not build TIDB_RESTRICTED_READ_ONLY upon read-only and super-read-only. These two variables will be remained untouched.
+
+We will create a new global variable TIDB_RESTRICTED_READ_ONLY, and turning it to on and off will only be allowed in SEM mode, and only users with RESTRICTED_VARIABLES_ADMIN will be allowed to modify it.
```

Comment:
This is contradicted in the source because of `RESTRICTED_REPLICA_WRITER_ADMIN`. Please clarify that replication traffic is also still permitted.

Reply:
sure